client/src/Components/Common/CommentContainer/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+import { List } from 'antd';
+import PropTypes from 'prop-types';
+import Comment from '../Comment';
+import MainButton from '../MainButton';
+
+const commentsPerPage = 5;
+let arrayForHoldingComments = [];
+
+const CommentContainer = ({ data }) => {
+  const [commentsToShow, setCommentsToShow] = useState([]);
+  const [next, setNext] = useState(5);
+
+  const loopWithSlice = (start, end) => {
+    const slicedComments = data.slice(start, end);
+    arrayForHoldingComments = [...arrayForHoldingComments, ...slicedComments];
+    setCommentsToShow(arrayForHoldingComments);
+  };
+
+  const handleShowMoreComments = () => {
+    loopWithSlice(next, next + commentsPerPage);
+    setNext(next + commentsPerPage);
+  };
+
+  useEffect(() => {
+    loopWithSlice(0, commentsPerPage);
+  }, []);
+
+  return (
+    <div>
+      <List
+        itemLayout="horizontal"
+        dataSource={commentsToShow}
+        renderItem={(item) => (
+          <List.Item>
+            <Comment
+              isAdmin={item.isAdmin}
+              userName={item.userName}
+              commentText={item.commentText}
+              date={item.date}
+              rateValue={item.rateValue}
+            />
+          </List.Item>
+        )}
+      />
+      {commentsToShow.length !== data.length ? (
+        <MainButton onClick={handleShowMoreComments}>
+          عرض المزيد من التعليقات
+        </MainButton>
+      ) : (
+        <MainButton backgroundColor="#ABC4C1">
+          عرض المزيد من التعليقات
+        </MainButton>
+      )}
```

Comment:
the button should appear just in case the ```commentsToShow``` length less  than ```data``` length otherwise doesn't make sense to show the button says show more and doesn't do anything, you have handled the ```handleShowMoreComments``` function and that's great but you need to fix the render condition here 
```suggestion
      {commentsToShow.length < data.length && (
        <MainButton onClick={handleShowMoreComments}>
          عرض المزيد من التعليقات
        </MainButton>
      )}
```